silk/tests/test_integration.py

```diff
@@ -0,0 +1,83 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import mock
+import pytest
+
+from datadog_checks.silk import SilkCheck
+
+from .common import BASE_TAGS, BLOCKSIZE_METRICS, HOST, METRICS, READ_WRITE_METRICS, SYSTEM_TAGS
+
+
+@pytest.mark.parametrize(
+    'enable_rw, enable_bs, expected_metrics',
+    [
+        pytest.param(False, False, [*METRICS], id="rw bs disabled"),
+        pytest.param(True, True, [*METRICS, *BLOCKSIZE_METRICS, *READ_WRITE_METRICS], id="rw bs enabled"),
+        pytest.param(False, True, [*METRICS, *BLOCKSIZE_METRICS], id="bs enabled"),
+        pytest.param(True, False, [*METRICS, *READ_WRITE_METRICS], id="rw enabled"),
+    ],
+)
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_check(dd_run_check, aggregator, instance, enable_rw, enable_bs, expected_metrics):
+    instance['enable_read_write_statistics'] = enable_rw
+    instance['enable_blocksize_statistics'] = enable_bs
+
+    check = SilkCheck('silk', {}, [instance])
+    dd_run_check(check)
+    for metric in expected_metrics:
+        aggregator.assert_metric(metric)
+        for tag in [*BASE_TAGS, *SYSTEM_TAGS]:
+            aggregator.assert_metric_has_tag(metric, tag)
+
+
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_error_msg_response(dd_run_check, aggregator, instance):
+    error_response = {"error_msg": "Statistics data is unavailable while system is OFFLINE"}
+    with mock.patch('datadog_checks.base.utils.http.requests.Response.json') as g:
+        g.return_value = error_response
+        check = SilkCheck('silk', {}, [instance])
+        dd_run_check(check)
+        aggregator.assert_service_check(
+            'silk.can_connect', SilkCheck.WARNING, message="Received error message: " + error_response["error_msg"]
+        )
+
+
+@pytest.mark.integration
+def test_incorrect_config(dd_run_check):
+    invalid_instance = {'host_addres': 'localhost'}  # misspelled required parameter
+    with pytest.raises(Exception):
+        check = SilkCheck('silk', {}, [invalid_instance])
+        dd_run_check(check)
```

Comment:
I think you don't need to run the check since the validation is in the `__init__` method, which means you can also specify the `ConfigurationError` because it won't be hidden by `dd_run_check`

```suggestion
    with pytest.raises(ConfigurationError):
        check = SilkCheck('silk', {}, [invalid_instance])
```